angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -69,6 +84,21 @@ define(['angular'], function(angular){
         });
       }
     };
+    
+    $scope.changePref=function(userPreference){
+
+    	if(userPreference==='C'){
+    		$scope.changeToC();
+    		$scope.showMetric=true;
+    	}else{
+    		$scope.changeToF();
```

Comment:
You could add in a `$scope.showMetric = false;` here and remove the double call in the `ng-click`